yt/utilities/amr_kdtree/amr_kdtools.py

```diff
@@ -3,34 +3,39 @@
 from yt.funcs import mylog
 
 
-def receive_and_reduce(comm, incoming_rank, image, add_to_front):
+def receive_and_reduce(comm, incoming_rank, image, add_to_front, use_opacity):
     mylog.debug("Receiving image from %04i", incoming_rank)
     # mylog.debug( '%04i receiving image from %04i'%(self.comm.rank,back.owner))
     arr2 = comm.recv_array(incoming_rank, incoming_rank).reshape(
         (image.shape[0], image.shape[1], image.shape[2])
     )
 
-    if add_to_front:
-        front = arr2
-        back = image
+    if use_opacity:
```

Comment:
We could reduce indentation and keep the diff size minimal by writing this new logic as an early return instead of moving everything into a branch
```suggestion
    if not use_opacity:
        np.add(image, arr2, image)
        return image
```